instrumentation/src/java/org/oppia/android/instrumentation/TestApplicationComponent.kt

```diff
@@ -0,0 +1,96 @@
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.ApplicationStartupListener
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.exceptions.UncaughtExceptionLoggerModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.platformparameter.PlatformParameterModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.util.accessibility.AccessibilityProdModule
+import org.oppia.android.util.caching.CachingModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.DebugLogReportingModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.system.OppiaClockModule
+import org.oppia.android.util.threading.DispatcherModule
+import javax.inject.Provider
+import javax.inject.Singleton
+
+/**
+ * Root Dagger component for the application. All application-scoped modules should be included in
+ * this component.
+ *
+ * At the time of building the app in prod mode -
+ * Remove: [DeveloperOptionsStarterModule], [DebugLogReportingModule]
+ * Add: [LogReportingModule]
+ */
+@Singleton
+@Component(
+  modules = [
+    ApplicationModule::class, DispatcherModule::class,
+    LoggerModule::class, OppiaClockModule::class,
+    ContinueModule::class, FractionInputModule::class,
+    ItemSelectionInputModule::class, MultipleChoiceInputModule::class,
+    NumberWithUnitsRuleModule::class, NumericInputRuleModule::class,
+    TextInputRuleModule::class, DragDropSortInputModule::class,
+    InteractionsModule::class, TestGcsResourceModule::class,
+    GlideImageLoaderModule::class, TestImageParsingModule::class,
+    HtmlParserEntityTypeModule::class, CachingModule::class,
+    QuestionModule::class, DebugLogReportingModule::class,
+    AccessibilityProdModule::class, ImageClickInputModule::class,
+    LogStorageModule::class, IntentFactoryShimModule::class,
+    ViewBindingShimModule::class, PrimeTopicAssetsControllerModule::class,
+    ExpirationMetaDataRetrieverModule::class, RatioInputModule::class,
+    UncaughtExceptionLoggerModule::class, ApplicationStartupListenerModule::class,
+    LogUploadWorkerModule::class, WorkManagerConfigurationModule::class,
+    HintsAndSolutionConfigModule::class, FirebaseLogUploaderModule::class,
+    TestNetworkModule::class, PracticeTabModule::class, PlatformParameterModule::class,
+    ExplorationStorageModule::class, DeveloperOptionsStarterModule::class,
+    DeveloperOptionsModule::class
```

Comment:
Hmm. This might actually make the need for smoke tests essential. As we change modules, the end-to-end tests will very easily break without detection if we don't at least have a basic build test running in CI.

Reply:
https://github.com/oppia/oppia-android/issues/3618
Created a issue to keep track of it.

To me it looks like we already have smoke tests accroding to this
https://github.com/oppia/oppia-android/pull/3505/checks?check_run_id=3208383668

Reply:
That's not quite the same (a smoke test would pass if it builds), but given the discussed scope for the project we'll work on adding these later.